homeassistant/components/spider/climate.py

```diff
@@ -41,8 +37,19 @@ class SpiderThermostat(ClimateEntity):
 
     def __init__(self, api, thermostat):
         """Initialize the thermostat."""
+        self.support_fan = []
+        self.support_hvac = []
         self.api = api
         self.thermostat = thermostat
+        for fan_speed_value in thermostat.fan_speed_values:
+            self.support_fan.append(fan_speed_value)
+        operation_values = thermostat.operation_values
+        for operation_value in operation_values:
+            self.support_hvac.append(
+                SPIDER_STATE_TO_HA.get(
+                    operation_value, lambda: "Invalid operation value"
```

Comment:
Should this be a `lambda`? Can `hvac_modes` deal with that? The ClimateEntity expects a `List[str]` containing a subset of `HVAC_MODES`.

Reply:
Removed lambda and moved conversion to HA value to hvac_modes method

See https://github.com/home-assistant/core/pull/44575/commits/f6a8ca81c7b6dad566bad653f767af86a125246e

Reply:
Building the list in `__init__` should be fine to avoid building it every time `hvac_modes` is called (it's not modified elsewhere anyway), but `"Invalid operation value"` is not in `HVAC_MODES` so it should not be in the list returned by `hvac_modes`.
I would suggest to log a warning for any invalid modes and omit it from the list.

See the docstring of `hvac_modes` on the parent class for reference:
https://github.com/home-assistant/core/blob/067f2d0098d13134284e442d75ae5f1015ae044c/homeassistant/components/climate/__init__.py#L295-L301

Reply:
Updated according to your suggestion